src/Controller/EventsController.php

```diff
@@ -733,16 +734,25 @@ public function rejectAction(Request $request, PDO $db)
 
         $event_id     = $this->getItemId($request);
         $event_mapper = new EventMapper($db, $request);
+        $reason = filter_var($request->getParameter('reason'), FILTER_SANITIZE_STRING);
 
         if (!$event_mapper->thisUserCanApproveEvents()) {
             throw new Exception("You are not allowed to reject this event", Http::FORBIDDEN);
         }
 
-        $result = $event_mapper->reject($event_id, $request->user_id);
+        $result = $event_mapper->reject($event_id, $request->user_id, $reason);
         if (!$result) {
             throw new Exception("This event cannot be rejected", Http::BAD_REQUEST);
         }
 
+        // Only send an email if we provided a reason
+        if (!empty($reason)) {
+            $event        = $event_mapper->getEventById($event_id, true, false)['events'][0];
+            $recipients   = array_merge([$this->config['email']['from']], $event_mapper->getHostsEmailAddresses($event_id));
+            $emailService = new EventRejectedEmailService($this->config, $recipients, $event);
```

Comment:
To me this looks like an instance of `EventRejectedEmailService` is a hard dependency of this controller action. 

Instead of creating it here, can we inject it via constructor injection?

Reply:
I get it, but we're doing the exact same thing with `EventApprovedEmailService`. I'd rather ship and iterate than get botted down in detail.

Reply:
So we prefer to continue shipping software with the same quality, rather than turning the ship around?